Dockerfile

```
@@ -0,0 +1,29 @@
+# syntax=docker/dockerfile:experimental
+
+FROM python:3.12.9-alpine as builder
+
+# create virtual environment
+RUN python -m venv /opt/venv
+ENV PATH="/opt/venv/bin:$PATH"
+
+COPY requirements.txt .
+RUN pip install --upgrade pip
```

Hadolint (2.12.0)

[warning] Pin versions in pip. Instead of `pip install <package>` use `pip install <package>==<version>` or `pip install --requirement <requirements file>`

(DL3013)